src/providers/smtp.rs

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::net::SocketAddr;
+
+use lettre::{smtp::error::Error as SmtpError, ClientSecurity, EmailTransport, SmtpTransport};
+use lettre_email::{error::Error as EmailError, EmailBuilder, Header as LettreHeader};
+
+use super::{Headers, Provider};
+use app_errors::{AppError, AppErrorKind, AppResult};
+use settings::Settings;
+
+pub struct SmtpProvider {
+    host: String,
+    port: u16,
+    sender: (String, String),
+}
+
+impl SmtpProvider {
+    pub fn new(settings: &Settings) -> SmtpProvider {
+        SmtpProvider {
+            host: settings.smtp.host.0.clone(),
+            port: settings.smtp.port,
+            sender: (
+                settings.sender.address.0.clone(),
+                settings.sender.name.0.clone(),
+            ),
+        }
+    }
+}
+
+impl Provider for SmtpProvider {
+    fn send(
+        &self,
+        to: &str,
+        cc: &[&str],
+        headers: Option<&Headers>,
+        subject: &str,
+        body_text: &str,
+        body_html: Option<&str>,
+    ) -> AppResult<String> {
+        let mut email = EmailBuilder::new()
+            .to(to)
+            .from(self.sender.clone())
+            .subject(subject)
+            .text(body_text);
+
+        for address in cc {
+            email = email.cc(*address);
+        }
+
+        if let Some(headers) = headers {
+            for (name, value) in headers {
+                email = email.header(LettreHeader::new(name.clone(), value.clone()));
+            }
+        }
+
+        if let Some(body_html) = body_html {
+            email = email.html(body_html);
+        }
+
+        let message = email.build()?;
+        let addr = SocketAddr::new(self.host.parse().unwrap(), self.port);
```

Comment:
I don't love this unwrap here, but I guess if we change to IpAddrr in the settings we wouldn't need it anyways, so Ill wait.